### PR TITLE
Created new Navigation-v2 component

### PR DESCRIPTION
**Pull request by lakhaNamdhari**
/cc @AutoScout24/web-experience

## Description
- The navigation has been re-written using flexbox
- the component is designed using mobile first approach
- The right navigation has been enhanced with Country selector
- Added two variations to demonstrate different supported layouts
- Modified navigation.js to instantiate more than one navigation on page

## Risks
- [medium] Added a new component but modified navigation.js to instantiate more than one navigation on page.

### CODE REVIEW

File: src/06-components/organisms/navigation-v2/docs/navigation-v2-default.html [hunk 1]
```diff
@@ -0,0 +1,103 @@
+<a class="skip-main" href="#main-target">Skip to main content</a>
+<header class="sc-navigation-v2" role="navigation">
+    <div class="bar">
+        <button class="sc-btn-mobile-menu" aria-expanded="false"><span></span>Menu</button>
+        <a href="https://www.autoscout24.de" class="icon-auto24">
+            <as24-icon type="auto24"></as24-icon>
+        </a>
+    </div>
+    <nav aria-labelledby="mainmenu">
```
[the remark this answers is not in the document]
Comment (rowantower) [reply src/06-components/organisms/navigation-v2/docs/navigation-v2-default.html]: Agree, shall we name attribute `aria-label` instead?

File: src/06-components/organisms/navigation-v2/navigation-v2.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        const keyCode = e.which;
+        const header = e.target.closest('.sc-navigation-v2');
+        if (!header) {
+            return;
+        }
+        if(keyCode === 9) {
+            // TAB
+            header.classList.add('sc-focus-visible-only');
+        } else if(keyCode === 27) {
+            // ESCAPE
```
[the remark this answers is not in the document]
Comment (rowantower) [reply src/06-components/organisms/navigation-v2/navigation-v2.js]: I am pro this feature, as in my opinion, it's expected behavior from a popover.
I will ask couple of our UX designers to clarify it

File: src/06-components/organisms/navigation-v2/navigation-v2.scss [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+      color: $orange-100;
+    }
+  }
+
+  li {
+    &:hover,
+    &.open {
+      .title {
+        &::after {
+          background-image: url('data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxMC41IiBoZWlnaHQ9IjYuMjc2NjEzMiIgdmlld0JveD0iMCAwIDEwLjUgNi4yNzY2MTMyIiB2ZXJzaW9uPSIxLjEiPjxwYXRoIGQ9Ik0gMC4wMDQzNjQ1OSwwLjAyODQ2Mzk3IDQuNTc4ODgzNCw0LjIxOTIwNzYgOS4xNTMyNTg5LDAuMDI4ODE5NzUiIHN0cm9rZS13aWR0aD0iMS45OTk2NzAzOSIgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoLjY3IC43MTQpIiBzdHJva2U9IiNmZjc1MDAiIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCIvPjwvc3ZnPg==');
```
[the remark this answers is not in the document]
Comment (rowantower) [reply src/06-components/organisms/navigation-v2/navigation-v2.scss]: I would like to, but then we need to use svgs. These two are the same arrow but in two colors (gray and orange)

File: src/06-components/organisms/navigation-v2/navigation-v2.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (rowantower) [reply src/06-components/organisms/navigation-v2/navigation-v2.js]: the short survey in UX channel showed that this feature is still appreciated

File: src/07-utilities/utils.scss [hunk 4]
```diff
@@ -372,4 +372,16 @@
     height: $XL - 1;
   }
 }
+
+.sc-focus-visible-only {
+  button:focus {
+    outline: 5px auto -webkit-focus-ring-color;
```
Comment (bdadam) [inline src/07-utilities/utils.scss]: `-webkit-focus-ring-color` only works in webkit/blink based browsers

Comment (rowantower) [reply src/07-utilities/utils.scss]: What if we add
outline: 5px auto;
outline: 5px auto -webkit-focus-ring-color;
and non webkit will get the fallback?

**Comment by rowantower** [comment]
I would really appreciate if we can merge it tomorrow morning ;)

**Pull request merged**